From cbd20f0cd9f5d60f6a05b224b993484d33a878fd Mon Sep 17 00:00:00 2001
From: Richard Acayan <mailingradian@gmail.com>
Date: Sat, 23 Sep 2023 21:04:23 -0400
Subject: [PATCH] drm/panel: samsung-sofef00-bonito: add enable function as
 callback

As reported by https://gitlab.com/77z.us, the panel does not turn on
without adding the enable function as a callback. Add it to fix the
supported panel on the Pixel 3a XL.

Amends: drm/panel: samsung-sofef00-bonito: move dsi commands to enable function
Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 drivers/gpu/drm/panel/panel-samsung-sofef00-bonito.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-samsung-sofef00-bonito.c b/drivers/gpu/drm/panel/panel-samsung-sofef00-bonito.c
index fdd7b20b1c78..a8bfded80d9c 100644
--- a/drivers/gpu/drm/panel/panel-samsung-sofef00-bonito.c
+++ b/drivers/gpu/drm/panel/panel-samsung-sofef00-bonito.c
@@ -51,7 +51,7 @@ static int sofef00_prepare(struct drm_panel *panel)
 	return 0;
 }
 
-static int sofef00_on(struct drm_panel *panel)
+static int sofef00_enable(struct drm_panel *panel)
 {
 	struct sofef00 *ctx = to_sofef00(panel);
 	struct mipi_dsi_device *dsi = ctx->dsi;
@@ -161,6 +161,7 @@ static int sofef00_get_modes(struct drm_panel *panel,
 
 static const struct drm_panel_funcs sofef00_panel_funcs = {
 	.prepare = sofef00_prepare,
+	.enable = sofef00_enable,
 	.disable = sofef00_disable,
 	.unprepare = sofef00_unprepare,
 	.get_modes = sofef00_get_modes,
-- 
2.42.0

